yt/data_objects/grid_patch.py

```diff
@@ -173,6 +181,20 @@ def _prepare_grid(self):
         self.ActiveDimensions = h.grid_dimensions[my_ind]
         self.LeftEdge = h.grid_left_edge[my_ind]
         self.RightEdge = h.grid_right_edge[my_ind]
+        # This can be expensive so we allow people to disable this behavior
+        # via a config option
+        if bool(ytcfg.get('yt', 'reconstruct_index')):
```

Comment:
Out of curiosity I profiled this part. The performance drop you're seeing comes from evaluating `ytcfg.get` 400000 times. If you apply something along those lines:

```
diff --git a/yt/data_objects/grid_patch.py b/yt/data_objects/grid_patch.py
index 61b586d2b..7d393c89e 100644
--- a/yt/data_objects/grid_patch.py
+++ b/yt/data_objects/grid_patch.py
@@ -69,6 +69,7 @@ class AMRGridPatch(YTSelectionContainer):
         self._last_selector_id = None
         self._current_particle_type = 'all'
         self._current_fluid_type = self.ds.default_fluid_type
+        self._reconstruct_index = bool(ytcfg.get('yt', 'reconstruct_index'))
 
     def get_global_startindex(self):
         """
@@ -183,7 +184,7 @@ class AMRGridPatch(YTSelectionContainer):
         self.RightEdge = h.grid_right_edge[my_ind]
         # This can be expensive so we allow people to disable this behavior
         # via a config option
-        if bool(ytcfg.get('yt', 'reconstruct_index')):
+        if self._reconstruct_index:
             if iterable(self.Parent) and len(self.Parent) > 0:
                 p = self.Parent[0]
             else:
```
it should be much better